### PR TITLE
Refs Task#384069 Create chat bot

### PR DESCRIPTION
**Pull request by tamtt-1031**
## Related Tickets

- [#384069](https://dev.sun-asterisk.com/issues/384069)

## What's this PR do ?

- [x] Implement Create chat bot

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/BotController.php [hunk 1]
```diff
@@ -22,4 +23,29 @@ public function index()
...
+
+    public function create()
+    {
+        return view('bots.create');
+    }
+
+    public function store(BotCreateRequest $request)
+    {
+        $data = $request->except('_token');
+        $data['token'] = md5(Auth::id().''.time());
```
Comment (phucqd-1305) [inline app/Http/Controllers/BotController.php]: nối thêm cái '' vào làm gì vậy em, chỉ cần md5(Auth::id() . time()) là được mà

File: app/Http/Requests/BotCreateRequest.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'name' => 'required|min:1|max:50|unique:bots,name,NULL,id,user_id,'.Auth::id(),
```
Comment (phucqd-1305) [inline app/Http/Requests/BotCreateRequest.php]: trước và sau dấu . nối phải có 1 dấu cách nhé, CI sao k báo lỗi nhỉ :vertical_traffic_light:

Comment (tamtt-1031) [reply app/Http/Requests/BotCreateRequest.php]: chắc do e ăn ở tốt. CI nó bỏ qua =))
Đùa chứ chắc file config của CI ko bắt lỗi này

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Comment by phucqd-1305** [comment]
Conflict rồi

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Review by nguyenthinhhn — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/BotController.php [hunk 3]
```diff
@@ -31,4 +34,28 @@ public function destroy($id)
...
+        try {
+            $bot = $this->botRepository->create($data);
+            return redirect()->route('bots.edit', $bot)
+                             ->with('messageSuccess', 'This bot successfully created');
+        } catch (QueryException $exception) {
+            return redirect()->back()->with('messageFail', 'Create failed. Something went wrong')->withInput();
+        }
+    }
+
+    public function edit(Bot $bot)
```
Comment (ghost) [inline app/Http/Controllers/BotController.php:57]: làm luôn update à a :D

Comment (tamtt-1031) [reply app/Http/Controllers/BotController.php:57]: ko e. sau khi success nó vào redirect tới thằng edit để xem lại thông á.
làm tạm cái hàm trước thôi. :D vài bữa làm edit thì nó chỉ show ra luôn ko cần phải vào chỉnh lại thằng create :D

**Review by ghost — APPROVED** [review]

**Review by quanghung97 — APPROVED** [review]
OK

**Comment by phucqd-1305** [comment]
ok

**Pull request merged**